optimum/intel/openvino/quantization.py

```diff
@@ -234,17 +235,26 @@ def __init__(self, request):
                 self.request = request
 
             def __call__(self, *args, **kwargs):
-                data_cache.append(*args)
+                data_cache.append(deepcopy(*args))
                 return self.request(*args, *kwargs)
 
+            def start_async(
+                self,
+                inputs: Any = None,
+                userdata: Any = None,
+                shared_memory: bool = False,
+            ) -> None:
+                data_cache.append(inputs)
+                self.request.start_async(inputs, userdata, shared_memory)
+
             def __getattr__(self, attr):
                 if attr in self.__dict__:
                     return getattr(self, attr)
                 return getattr(self.request, attr)
 
         self.model.request = InferRequestWrapper(self.model.request)
-        for i, data in enumerate(calibration_dataloader):
-            self.model.generate(**data, max_new_tokens=10)
+        for _, data in enumerate(calibration_dataloader):
```

Comment:
Not related to the PR, but what do you think about uniformizing how quantization is applied on causal langage models depending on whether the user gives a `torch.nn.Module` or a `OVBaseDecoderModel` (the number of generation steps is currently not the same). We could also instantiate an `OVModel` in the `from_pretrained` method when the given model is a `PreTrainedModel`

Reply:
It is hard to accomplish this with the current NNCF PTQ API implementation we have for PyTorch. I think we should deprecate PTQ for PyTorch at some point because it also introduces ambiguity for the user about what workflow to use for quantization.